operator/rpc_client.go

```diff
@@ -164,39 +200,42 @@ func (c *AggregatorRpcClient) tryResendFromDeque() {
 }
 
 func (c *AggregatorRpcClient) sendOperatorMessage(sendCb func() error, message RpcMessage) {
+	c.rpcClientLock.RLock()
```

Comment:
I don't think we're risking a deadlock here in the way the unsent messages lock is used, and in terms of `handleRpcError`, it uses a separate routine for closing the client, so no double lock either.
It's needed because we don't want the client to be destroyed during this call.

Reply:
My bad, didn't notice goroutine in `handleRpcError`